### PR TITLE
Arrow node implementation

### PR DESCRIPTION
**Pull request by printfCalvin**
- Switching kuzu's own node CSV reader to apache arrow.
- Fixing test files to follow the CSV standard.
- Fixing assertion failure when reading an empty list.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,15 @@ Kùzu is an in-process property graph database management system (GDBMS) built f
 Kùzu is being actively developed at University of Waterloo as a feature-rich and usable GDBMS. Kùzu is available under a permissible license. So try it out and help us make it better! We welcome your feedback and feature requests.
 
 ## Build
+<<<<<<< Updated upstream
```
Comment (ray6080) [inline README.md:20]: Clean up these.

File: dataset/copy-csv-node-property-test/vPerson.csv [hunk 2]
```diff
@@ -6,7 +6,15 @@
 5,"qifidjufri"
 6,"gqpnpbdmrb"
 7,"dgzbiqjkaz"
+<<<<<<< Updated upstream
```
Comment (ray6080) [inline dataset/copy-csv-node-property-test/vPerson.csv:9]: Ditto.

File: dataset/tinysnb/vPerson.csv [hunk 3]
```diff
@@ -1,4 +1,6 @@
 id,fname,Gender,ISStudent,isWorker,age,eyeSight,birthdate,registerTime,lastJobDuration,workedHours,usedNames,courseScoresPerTerm
+<<<<<<< Updated upstream
```
Comment (ray6080) [inline dataset/tinysnb/vPerson.csv:2]: Ditto.

File: src/binder/bind/bind_ddl.cpp [hunk 4]
```diff
@@ -3,7 +3,13 @@
 #include "binder/bind/bound_drop_table.h"
 #include "binder/binder.h"
 #include "parser/ddl/create_node_clause.h"
+<<<<<<< Updated upstream
```
Comment (ray6080) [inline src/binder/bind/bind_ddl.cpp:6]: Did you even compile this locally before opening a PR???

**Review by ray6080 — CHANGES_REQUESTED** [review]
There are many unresolved messages regarding to merging/squashing. Can you clean those first? I will look into the code after you clean those out.

**Review by ray6080 — APPROVED** [review]

File: benchmark/Dockerfile [hunk 5]
```diff
@@ -3,8 +3,11 @@ FROM ubuntu:22.04
 ENV CSV_DIR /csv
 ENV SERIALIZED_DIR /serialized
 
+RUN wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
```
Comment (mewim) [inline benchmark/Dockerfile:6]: The Dockerfile does not seem to be reverted correctly. @ray6080 could you please revert this before merging the PR?

**Review by mewim — CHANGES_REQUESTED** [review]
The Dockerfile does not seem to be reverted correctly. @ray6080 could you please revert this before merging the PR?

**Review by mewim — APPROVED** [review]

File: benchmark/Dockerfile (hunk 5 above)
Comment (ray6080) [reply benchmark/Dockerfile:6]: Yeah. nice catch. thanks!
I changed it now.

**Pull request merged**